### PR TITLE
Socketio

### PR DESCRIPTION
**Pull request by AnthonyHewins**
Here's basically everything we will need for socketio.

I wasn't able to test it completely, but I didn't want to block anyone else's progress so here's the code as of right now.

### CODE REVIEW

**Review by nlferman — COMMENTED** [review]
This all looks pretty good on my end. Josh or Anthony, would it be possible to get one room set up so I can test connecting to a room as well as test when a room isn't found?

File: server/models.py [hunk 1]
```diff
@@ -74,6 +75,13 @@ class GameMap(Document):
     def save(self, *args, **kwargs):
         self.updated = datetime.now()
         super(GameMap, self).save(*args, **kwargs)
+        all_open_sessions = Session.get(map=self.id)
+
+        # Unsure if it's possible to have multiple sessions
+        # with the same map, (going to guess this is a no)
+        # so loop may be unneeded code
+        for i in all_open_sessions:
+            somesockets.update(self.id, i.code)
```
Comment (joshherkness) [inline server/models.py:84]: Probably not, but it is a good idea to handle this case here.

**Review by joshherkness — APPROVED** [review]

**Comment by joshherkness** [comment]
@goeteeks You still don't have the node client up and running? That would be used to test this. I think Kajuan is trying to setup a docker container so that we can all run every aspect of the project.

**Comment by joshherkness** [comment]
There were a bunch of breaking changes with the current development branch. I've reverted the merge into master. Can you fix this up and create another PR when you are done?

**Comment by joshherkness** [comment]
@AnthonyHewins

**Pull request merged**